Python/compile.c

```diff
@@ -4902,22 +4902,25 @@ struct assembler {
 };
 
 static void
-dfs(struct compiler *c, basicblock *b, struct assembler *a)
+dfs(struct compiler *c, basicblock *b, struct assembler *a, int nblocks)
 {
-    int i;
-    struct instr *instr = NULL;
+    int i, j;
 
-    if (b->b_seen)
-        return;
-    b->b_seen = 1;
-    if (b->b_next != NULL)
-        dfs(c, b->b_next, a);
-    for (i = 0; i < b->b_iused; i++) {
-        instr = &b->b_instr[i];
-        if (instr->i_jrel || instr->i_jabs)
-            dfs(c, instr->i_target, a);
+    /* Get rid of recursion for normal control flow. */
+    for (j = nblocks; b && !b->b_seen; b = b->b_next) {
+        b->b_seen = 1;
+        assert(a->a_nblocks < j);
+        a->a_postorder[--j] = b;
```

Comment:
This only works because blocks are emitted in topological order, right? It would be nice to add a comment.